src/OpenTK.Windowing.Desktop/NativeWindow.cs

```diff
@@ -789,47 +773,41 @@ private unsafe void HandleResize(int width, int height)
 
         private unsafe void RegisterWindowCallbacks()
         {
-            _posCallback = (window, x, y) => OnMove(new WindowPositionEventArgs(x, y));
-            GLFW.SetWindowPosCallback(WindowPtr, _posCallback);
+            GLFW.SetWindowPosCallback(WindowPtr, (w, x, y) => OnMove(new WindowPositionEventArgs(x, y)));
+
+            GLFW.SetWindowSizeCallback(WindowPtr, (w, width, height) => OnResize(new ResizeEventArgs(width, height)));
 
-            _sizeCallback = (window, width, height) => OnResize(new ResizeEventArgs(width, height));
-            GLFW.SetWindowSizeCallback(WindowPtr, _sizeCallback);
+            GLFW.SetWindowCloseCallback(WindowPtr, OnCloseCallback);
 
-            _closeCallback = OnCloseCallback;
-            GLFW.SetWindowCloseCallback(WindowPtr, _closeCallback);
+            GLFW.SetWindowIconifyCallback(WindowPtr, (w, iconified) => OnMinimized(new MinimizedEventArgs(iconified)));
 
-            _iconifyCallback = (window, iconified) => OnMinimized(new MinimizedEventArgs(iconified));
-            GLFW.SetWindowIconifyCallback(WindowPtr, _iconifyCallback);
+            GLFW.SetWindowFocusCallback(WindowPtr, (w, focused) => OnFocusedChanged(new FocusedChangedEventArgs(focused)));
 
-            _focusCallback = (window, focused) => OnFocusedChanged(new FocusedChangedEventArgs(focused));
-            GLFW.SetWindowFocusCallback(WindowPtr, _focusCallback);
+            GLFW.SetCharCallback(WindowPtr, (w, codepoint) => OnTextInput(new TextInputEventArgs((int)codepoint)));
 
-            _charCallback = (window, codepoint) => OnTextInput(new TextInputEventArgs((int)codepoint));
-            GLFW.SetCharCallback(WindowPtr, _charCallback);
+            GLFW.SetKeyCallback(WindowPtr, KeyCallback);
 
-            _keyCallback = KeyCallback;
+            GLFW.SetCursorEnterCallback(WindowPtr, CursorEnterCallback);
 
-            GLFW.SetKeyCallback(WindowPtr, _keyCallback);
+            GLFW.SetMouseButtonCallback(WindowPtr, MouseButtonCallback);
 
-            _cursorEnterCallback = CursorEnterCallback;
-            GLFW.SetCursorEnterCallback(WindowPtr, _cursorEnterCallback);
+            GLFW.SetCursorPosCallback(WindowPtr, CursorPosCallback);
```

Comment:
There was a reason all of the callback delegates where stored as fields. 
If a reference to the delegate itself is not kept it will result in the delegate being garbage collected while it is still being referenced by native code, this will lead to a `ExecutionEngineException`.

Reply:
Solved.